AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/QtPyScriptCanvasNodePalette.py

```diff
@@ -30,18 +30,26 @@ def search_for_node(self, node_name: str):
 
         params node_name: the name of the node you are searching for
 
-        returns: True if the node_name we are searching for exists in the filtered search results
+        returns: a list of nodes found using the provided node_name string
         """
-        # Set the search field to blank text to refresh the visible node list
-        self.node_palette_search_box.setText("")
+        self.clear_search_filter()
         helper.wait_for_condition(lambda: self.node_palette_search_box.text() == node_name, WAIT_TIME_SEC_3)
 
         self.node_palette_search_box.setText(node_name)
         helper.wait_for_condition(lambda: self.node_palette_search_box.text() == node_name, WAIT_TIME_SEC_3)
```

Comment:
This first wait for condition will always wait 3 seconds since it's comparing to `node_name`, which hasn't been set into the search field yet.

I'm assuming you wanted this to compare that the filter has been cleared, and if so, should go into the `clear_search_filter` helper.

Otherwise, this is an erroneous step.

Reply:
yep. made a mistake and it was probably causing the test to lag a few seconds. Set the first wait comparison to check for a blank string.

Reply:
Please make sure you put the validation into the helper function too :)

Reply:
added a little bug armor here as well. these two methods will now check if the search field text is what we intended to put there and assert if not.